More cameras and fixes from darktable

pedrocr opened the pull request:
This adds a few cameras and adjusts others. I've also disabled all the NakedDecoder cameras that I have verified do not work yet or don't have files to test with.

klauspost commented on data/cameras.xml:
```diff
@@ -681,7 +711,7 @@
 			<Horizontal y="4" height="44"/>
 		</BlackAreas>
 	</Camera>
-	<Camera make="Canon" model="Canon EOS M2">
```
There is no need for this

pedrocr replied:
Yeah, we had a bunch of these from cameras that originally where ="no". I've removed them all now as it's just noise.

klauspost commented:
Note that I fixed a bug in the naked decoder, that broke all naked definitions with offset: https://github.com/klauspost/rawspeed/commit/f0668bfb6b373cc425732df9d2a7d954e4c74562

I don't like disabling cameras because we don't have samples for them. How many were broken, and how many without samples? I would rather just have a note saying "unconfirmed" on the ones without sample.

klauspost commented on data/cameras.xml:
```diff
@@ -491,7 +521,7 @@
 		</Hints>
 	</Camera>
-	<Camera make="Canon" model="Canon EOS 5DS R" decoder_version="6">
+	<Camera make="Canon" model="Canon EOS 5DS R" decoder_version="6" supported="no">
```
Why is this unsupported? I think I fixed that camera recently.

pedrocr replied:
The rawspeed support does seem fine but since there are no final images we didn't want to enable it yet. I guess that's a darktable thing, I'll remove it from the commit.

LibRaw replied:
Please note, that not only darktable uses RawSpeed, but other software uses it too.
Any LibRaw-based software (dozens directly and many indirectly, via FreeImage or other wrappers) may use RawSpeed as decoder.

Disabling right decoded camera in master branch because one library user (darktable) cannot process data  it (but why? this is very common CFA bayer) - IS A VERY BAD IDEA.

pedrocr commented:
Of those I disabled I only have samples for the minolta rd-175 and the Kodak c643, both are broken, both because they have weird color formats (the minolta because of the 3CCD setup the Kodak because of 422/420 color). In darktable we like to keep these things disabled so that when a file does show up and we add support in other places (color matrices, etc) there's no risk users who have used the broken support before will see their files start to return different results. I'll take that out of the patch if you prefer.

klauspost commented:
If it delivers unusable junk, it should be disabled, but if the data is essentially ok, not.

However, if you just want to disable it because darktable currently doesn't support it, you can use the `CameraMetaData::disableCamera( string make, string model)` to disable cameras selectively. That way you don't have to maintain a separate xml file. Adding support for a camera will already be a code change in darktable, so having this 'disabled list' shouldn't be a huge issue.

pedrocr commented:
Both cases could potentially be turned into a proper image with some post-processing but that code should really be in rawspeed. For the minolta case at least I'll probably port the dcraw code for it.

As for darktable support, since we already ship the cameras.xml file it's just easier to set that tag there than to have to ship additional code to create a blacklist. The 5Dr case is our own issue but the chdk ones with no samples I'd just disable in rawspeed as well. Unless we can find a sample just enabling them blind seems like a bad idea. Dave coffin supposedly has a sample file collection for all of these that he sells. I suppose he wouldn't be open to donating them to a "competing" open source project.

LibRaw commented:
PLEASE DO NOT THINK THAT DARKTABLE IS THE ONLY SOFTWARE THAT USES RawSpeed.
PLEASE!

klauspost commented:
@pedrocr: I think for your own sanity, a blacklist of your own would be easier for you to maintain, and it is clearer that all cameras present there is "supported by RawSpeed, but not by dt".

LibRaw commented:
More constructive:
 Minolta RD175 uses separate decoder in dcraw, it need to be implemented in RawSpeed to support this camera (I do not think it is worth it)
 For Kodak C603/643 - there are a lot of samples on rawsamples.ch with different encoding methods. Again, to list this camera in supported list it is better to implement all decoders, or simply drop the camera. Again, this camera do not worth it IMO.

But... 5Dr should not be disabled from RawSpeed master!

pedrocr commented:
@klauspost that's only the 5Dr and I already mentioned that I will be removing it from the diff. It's a temporary thing too so there's no real need to keep a separate blacklist. In general if we have a sample and rawspeed supports it we support it also.

@LibRaw I don't think that as I've pointed out in other comments. Please be civil, all caps comments are not nice and please read what I wrote, I've already addressed all your points.

LibRaw commented:
@pedrocr, sorry, I was excited. Will never repeat.

But I was really upset that 5DS(R) offered to be dropped because of other project problems. It means that I should maintain own RawSpeed branch and read all patches in upstream to not allow silent supported camera removal. It's not something which I would like to have

pedrocr commented:
I've reworked the diff. The 5Dr is no longer disabled and all the supported="yes" lines are gone. I've kept the NakedDecoder supported="no" lines as most of them will probably be wrong without some further tweaks. If you want those removed as well let me know.

klauspost commented:
I too am a bit torn about disabling the chdk cameras, I don't have the samples to confirm them, but on the other hand at most of them _should_ work reasonably well.

@LibRaw - do you have any opinion on this?

The other changes are fine, if you'd like you can take the chdk ones into a separate PR.

pedrocr commented:
Your call, I'll prepare the PR as preferred. The only reason they are there was that script you wrote to create them from the dcraw code. Seems like a waste to remove them (so we have the defs if a sample shows up) but I'd prefer not to have enabled cameras that are not verified to work. It makes it harder to do things like auto-generate supported cameras lists.

LibRaw commented:
From LibRaw project point of view, it does not matter, is RawSpeed support these 'CHDK' (really - many different types) cameras or not.
All (most?) of these cameras are supported by LibRaw, most of these files are not compressed (so decoding speed is acceptable without calling RawSpeed decoder) or very small (again, no need to use advanced decoder).

pedrocr commented:
If at all possible I'd prefer if this was merged as is. It would make it much easier not to have to keep this difference around between the darktable cameras.xml and the upstream one. Either way I'm waiting on this being resolved so I can send a couple of other PRs so it would be great if we could just decide either way.

klauspost commented:
Sorry for delaying this. Is it possible for you to send the chdk changes in a separate PR, at least then we can merge this one right away.

pedrocr committed:
Sync cameras from darktable

pedrocr commented:
I've pushed a version without the CHDK stuff. At least the RD175 is indeed broken but I'll put all the CHDK changes in a different PR as you wanted.

klauspost commented:
Thank you very much!
